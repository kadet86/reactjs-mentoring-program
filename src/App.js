import React, {Component, PureComponent} from 'react';

const noJsxElement = React.createElement('h3', {style: {color: "red"}}, 'Hello from no jsx element');

class HelloPure extends PureComponent {
    render() {
        return <h3 {...this.props}>Hello from pure component</h3>;
    }
}

function HelloFunc(props) {
    return <h3 {...props}>Hello from functional component</h3>;
}

export default class App extends Component {
    render() {
        return (
            <section>
                <h1>App Component header</h1>
                {noJsxElement}
                <HelloPure style={{color: "orange"}} />
                <HelloFunc style={{color: "green"}} />
            </section>
        );
    }
}
