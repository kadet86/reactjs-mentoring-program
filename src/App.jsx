import React from 'react';
import ErrorBoundary from './ErrorBoundary';
import FilteredMovieList from './FilteredMovieList';
import MoviePage from './MoviePage';

import { BrowserRouter as Router, Route, Redirect, Switch } from 'react-router-dom';

import './main.css';
import 'primereact/resources/themes/nova-light/theme.css';
import { NotFound } from './NotFound';

const App = () => {
    return (
        <ErrorBoundary>
            <main className="app">
                <Router>
                    <Switch>
                        <Route path="/" exact render={() => <Redirect to="/search" />} />
                        <Route path="/search/:query?/:searchBy?/:sortBy?" component={FilteredMovieList} />
                        <Route path="/film/:id" component={MoviePage} />
                        <Route component={NotFound} />
                    </Switch>
                </Router>
            </main>
        </ErrorBoundary>
    );
}

export default App;