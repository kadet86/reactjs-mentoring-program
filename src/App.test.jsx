import React from 'react';
import App from './App';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

describe('App component', () => {
    beforeEach(() => {
        fetch.resetMocks();
        fetch.once(JSON.stringify({ data: [{ genres: ['action'], id: 1 }] }));
    });

    it('shows 404 view for invalid path', () => {
        const { store } = createTestStore();
        const component = createComponent(store, '/invalid-path');
        expect(component.html()).toMatchSnapshot();
    });

    it('shows no results view for / path', () => {
        const { store } = createTestStore();
        const component = createComponent(store, '/', true);
        expect(component.html()).toMatchSnapshot();
    });

    it('shows filtered movie list for /search/:query path', () => {
        const movies = [{ genres: ['action'], id: 1 }];
        const { store } = createTestStore({ movies });
        const component = createComponent(store, '/search/green', true);
        expect(component.html()).toMatchSnapshot();
    });

    it('shows movie page for /film/:id path', () => {
        const movies = [{ genres: ['action'], id: 1 }];
        const movie = {
            genres: ['action'],
            title: 'action movie',
            id: 1,
        };

        fetch.once(JSON.stringify(movie));
        fetch.once(JSON.stringify({ data: movies }));

        const { store } = createTestStore({ movie, movies });
        const component = createComponent(store, '/film/1', true);

        expect(component.html()).toMatchSnapshot();
    });

    function createComponent(store, path, doMount = false) {
        const renderer = doMount ? mount : shallow;
        const component = renderer(
            <Provider store={store}>
                <MemoryRouter initialEntries={[path]}>
                    <App />
                </MemoryRouter>
            </Provider>
        );

        return component;
    }
});
