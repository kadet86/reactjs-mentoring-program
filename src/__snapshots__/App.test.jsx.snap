// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component shows 404 view for invalid path 1`] = `"<main class=\\"app\\"><main><h1>Page not found</h1></main></main>"`;

exports[`App component shows filtered movie list for /search/:query path 1`] = `"<main class=\\"app\\"><section class=\\"filtered-movie-list\\"><section class=\\"top-section\\"><div class=\\"top-section__overlay\\"></div><div class=\\"top-section__inner-container\\"><h1>FIND YOUR MOVIE</h1><form class=\\"search-form\\"><input class=\\"p-inputtext p-component search-input p-filled\\" placeholder=\\"Search...\\" type=\\"text\\" required=\\"\\" value=\\"green\\"><div class=\\"search-by\\"><label>SEARCH BY</label><button type=\\"button\\" class=\\"p-button p-component p-button-danger p-button-text-only\\" data-value=\\"title\\"><span class=\\"p-button-text p-c\\">Title</span></button><button type=\\"button\\" class=\\"p-button p-component p-button-secondary p-button-text-only\\" data-value=\\"genres\\"><span class=\\"p-button-text p-c\\">Genre</span></button></div><button class=\\"p-button p-component p-button-danger search-button p-button-text-only\\" type=\\"submit\\"><span class=\\"p-button-text p-c\\">Search</span></button></form></div></section><div class=\\"p-toolbar p-component movie-list-toolbar\\"><label class=\\"movie-list-toolbar__label\\">1 movies found</label><div class=\\"sort-by\\"><label>Sort by</label><a class=\\"sort-by__link sort-by__link--active\\" data-value=\\"release_date\\" href=\\"javascript:void(0);\\">release date</a><a class=\\"sort-by__link\\" data-value=\\"vote_average\\" href=\\"javascript:void(0);\\">rating</a></div></div><section class=\\"movie-list\\"><div class=\\"movie-list-item\\"><a href=\\"/film/1\\"><img></a><div><h2></h2><span class=\\"movie-list-item__date\\"></span></div><div class=\\"movie-list-item__genres\\">action</div></div></section></section></main>"`;

exports[`App component shows movie page for /film/:id path 1`] = `"<main class=\\"app\\"><section class=\\"movie-page\\"><section class=\\"top-section\\"><div class=\\"top-section__overlay\\"></div><div class=\\"top-section__inner-container\\"><button class=\\"p-button p-component p-button-secondary p-button-text-only\\"><span class=\\"p-button-text p-c\\">SEARCH</span></button><article class=\\"movie\\"><img><div class=\\"movie__description\\"><h2>action movie</h2>&nbsp;<div class=\\"movie__rating\\"></div><div class=\\"movie__tagline\\"></div><span class=\\"movie__date\\"></span><span class=\\"movie__duration\\">0 min</span><div class=\\"movie__overview\\"></div></div></article></div></section><section><div class=\\"p-toolbar p-component movie-list-toolbar\\"><label class=\\"movie-list-toolbar__label\\">Films by action genre</label></div><section class=\\"movie-list\\"><div class=\\"movie-list-item\\"><a href=\\"/film/1\\"><img></a><div><h2></h2><span class=\\"movie-list-item__date\\"></span></div><div class=\\"movie-list-item__genres\\">action</div></div></section></section></section></main>"`;

exports[`App component shows no results view for / path 1`] = `"<main class=\\"app\\"><section class=\\"filtered-movie-list\\"><section class=\\"top-section\\"><div class=\\"top-section__overlay\\"></div><div class=\\"top-section__inner-container\\"><h1>FIND YOUR MOVIE</h1><form class=\\"search-form\\"><input class=\\"p-inputtext p-component search-input\\" placeholder=\\"Search...\\" type=\\"text\\" required=\\"\\" value=\\"\\"><div class=\\"search-by\\"><label>SEARCH BY</label><button type=\\"button\\" class=\\"p-button p-component p-button-danger p-button-text-only\\" data-value=\\"title\\"><span class=\\"p-button-text p-c\\">Title</span></button><button type=\\"button\\" class=\\"p-button p-component p-button-secondary p-button-text-only\\" data-value=\\"genres\\"><span class=\\"p-button-text p-c\\">Genre</span></button></div><button class=\\"p-button p-component p-button-danger search-button p-button-text-only\\" type=\\"submit\\"><span class=\\"p-button-text p-c\\">Search</span></button></form></div></section><div class=\\"p-toolbar p-component\\"></div><section class=\\"no-results\\"><h1>No films found</h1></section></section></main>"`;
